plugins/UsersManager/Controller.php

```diff
@@ -405,6 +409,22 @@ public function recordUserSettings()
         return $toReturn;
     }
 
+    public function newsletterSignup()
+    {
+        Piwik::checkUserIsNotAnonymous();
+
+        $userLogin = Piwik::getCurrentUserLogin();
+        $email = Piwik::getCurrentUserEmail();
+
+        $success = NewsletterSignup::signupForNewsletter($userLogin, $email, true);
+        if ($success) {
+            $result = array('success' => true);
+        } else {
+            $result = array('error' => true);
+        }
+        return json_encode($result);
```

Comment:
You'll also need to call `Piwik\DataTable\Renderer\Json::sendHeaderJSON()` here since controllers will assume the output is `text/html`.

Reply:
Done